WSRequestHandler.cpp

```diff
@@ -1343,6 +1345,141 @@ void WSRequestHandler::HandleSetSceneItemCrop(WSRequestHandler* req) {
+        obs_data_set_double(scale_data, "x", scale.x);
+        obs_data_set_double(scale_data, "y", scale.y);
+        obs_data_set_obj(data, "scale", scale_data);
+
+        obs_data_t* crop_data = obs_data_create();
+        obs_sceneitem_crop crop;
+        obs_sceneitem_get_crop(scene_item, &crop);
+        obs_data_set_int(crop_data, "left", crop.left);
+        obs_data_set_int(crop_data, "top", crop.top);
+        obs_data_set_int(crop_data, "right", crop.right);
+        obs_data_set_int(crop_data, "bottom", crop.bottom);
+        obs_data_set_obj(data, "crop", crop_data);
+
+        obs_data_set_bool(data, "visible", obs_sceneitem_visible(scene_item));
+
+        obs_data_t* bounds_data = obs_data_create();
+        obs_bounds_type bounds_type = obs_sceneitem_get_bounds_type(scene_item);
+        if (bounds_type == OBS_BOUNDS_NONE) {
+            obs_data_set_bool(data, "bounds", false);
+        }
+        else {
+            switch(bounds_type) {
+                case OBS_BOUNDS_STRETCH: {
+                    obs_data_set_string(bounds_data, "type", "Stretch");
+                    break;
+                }
+                case OBS_BOUNDS_SCALE_INNER: {
+                    obs_data_set_string(bounds_data, "type", "Inner");
+                    break;
+                }
+                case OBS_BOUNDS_SCALE_OUTER: {
+                    obs_data_set_string(bounds_data, "type", "Outer");
+                    break;
+                }
+                case OBS_BOUNDS_SCALE_TO_WIDTH: {
+                    obs_data_set_string(bounds_data, "type", "Width");
+                    break;
+                }
+                case OBS_BOUNDS_SCALE_TO_HEIGHT: {
+                    obs_data_set_string(bounds_data, "type", "Height");
+                    break;
+                }
+                case OBS_BOUNDS_MAX_ONLY: {
+                    obs_data_set_string(bounds_data, "type", "Max");
+                    break;
+                }
+            }
+
+            // same with alignment above, decide if it should convert to string
+            // #define OBS_ALIGN_CENTER (0)
+            // #define OBS_ALIGN_LEFT   (1<<0)
+            // #define OBS_ALIGN_RIGHT  (1<<1)
+            // #define OBS_ALIGN_TOP    (1<<2)
+            // #define OBS_ALIGN_BOTTOM (1<<3)
+            obs_data_set_int(bounds_data, obs_sceneitem_get_bounds_alignment(scene_item));
+            vec2 bounds;
+            obs_sceneitem_get_bounds(scene_item, &bounds);
+            obs_data_set_float(bounds_data, "width", bounds.x);
+            obs_data_set_float(bounds_data, "height", bounds.y);
+            obs_data_set_obj(data, "bounds", bounds_data);
+        }
+
+        // add source width?
+        // add source height?
+        // add locked?
+
+        obs_sceneitem_release(scene_item);
+        req->SendOKResponse(data);
+    } else {
+        req->SendErrorResponse("specified scene item doesn't exist");
+    }
```

Comment:
Should i put this else at the top in the if as a pseudo "early return"

Reply:
I kind of like the idea of putting it at the top for consistency. It would match the other returns for invalid params, scene, etc.

At some point I'd like to have a look at extracting those methods out since they're just getting rewritten over and over again, but that's something for another day.

Reply:
I just wish I'd have better experience with c++ development before.
Visual Studio is a pain and I don't think I've ever had an easy time using makefiles. The only time I've actually been able to get builds + dll swapping going is at the office, but that's only because they have all the directories hardcoded and tell everyone to use the same installation directory structure.

Reply:
I'm a bit out of practice but in principle the changes I'm working on are pretty easy to make.
Half the battle with programming is environment and setup IMO.